objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/imageRetriever/FileRetriever.java

```diff
@@ -162,31 +156,39 @@ public ImageRetrieverResults getImage() throws ImageAcquisitionException {
             val += frameInterval;
             capture.set(Videoio.CAP_PROP_POS_FRAMES, val);
             
-            MatOfByte matOfByte = new MatOfByte();
-            // Translate the image into jpeg, error out if it cannot
-            if (!Imgcodecs.imencode(".jpg", matrix, matOfByte)) {
-                matOfByte.release();
-                matrix.release();
-                throw new ImageAcquisitionException(this.getClass().getCanonicalName() + ".mainVideoConversionError: " 
-                        + "Could not convert frame #" + val + " from video '" + imageLocation + "' into a jpeg image");
+            // Translate the image to jpeg
+            byte[] imageBytes = convertToJpeg(matrix);
+            if (imageBytes == null) {
+                throw new ImageAcquisitionException(this.getClass().getCanonicalName() + ".videoConversionError: " 
+                        + "Could not convert frame #" + val + " from video '" + defaultImageLocation 
+                        + "' into a jpeg image");
             }
-            byte [] imageByte = matOfByte.toArray();
-            matOfByte.release();
-            matrix.release();
-            
             
             otherData.put("frame", val);
-            results.setImage(imageByte);
+            results.setImage(imageBytes);
                     
             return results;
-        } else if (defaultImageFile != null){
-            try {
-                results.setImage(Files.readAllBytes(defaultImageFile.toPath()));
-                return results;
-            } catch (IOException e) {
-                throw new ImageAcquisitionException(this.getClass().getCanonicalName() + ".defaultImageReadError: " 
-                        + "Could not read '" + defaultImageFile.toPath() + "'");
+        } else if (defaultImageLocation != null){
+            // Read the expected image
+            otherData.put("file", defaultImageLocation);
+            Mat image = Imgcodecs.imread(defaultImageLocation);
+            if (image == null || image.empty()) {
+                if (image != null) {
+                    image.release();
+                }
+                throw new ImageAcquisitionException(this.getClass().getCanonicalName() + ".defaultImageUnreadable: " 
+                        + "Could not read requested file '" + defaultImageLocation + "'. "
+                        + "Most likely the image did not exist or was in an unreadable format");
```

Comment:
I can (and probably should) check if the file exists. The file type is a bit trickier, since OpenCV doesn't tell us and the file types could (but shouldn't) be different than the file extension. I could use ImageIO (though OpenCV and ImageIO might understand different types) or just assume the file extension is correct.

Reply:
Not worth doing that just to resolve an error.  Just make sure that the comments that say "the most likely reason..." include the file's not being where we expect it.  That'll be fine.